CSC8503CoreClasses/PhysicsSystem.cpp

```diff
@@ -202,6 +234,70 @@ so that objects separate back out.
 */
 void PhysicsSystem::ImpulseResolveCollision(GameObject& a, GameObject& b, CollisionDetection::ContactPoint& p) const {
 
+	PhysicsObject* physA = a.GetPhysicsObject();
+	PhysicsObject* physB = b.GetPhysicsObject();
+
+	Transform& transformA = a.GetTransform();
+	Transform& transformB = b.GetTransform();
+
+	float cRestitution = physA->GetElasticity() / physB->GetElasticity();
```

Comment:
Dividing coefficients of restitution isn't commutative, so the order of collision processing might make physics unreliable. Probably best to change this to something that is commutative: multiplying, averaging etc.

Reply:
Noted, can get an average done. This was done for the coursework because it needed that, will replace it by the average of the two.
Cheers.